coref/config.py

```diff
@@ -48,24 +52,43 @@ class Config:  # pylint: disable=too-many-instance-attributes, too-few-public-me
     bce_loss_weight: float
 
     tokenizer_kwargs: Dict[str, dict]
+
+    model_bank: ModelConfig = field(init=False)
+
     conll_log_dir: str
 
     def __post_init__(self):
-        with open(Path(self.train_data), "r") as f:
+        with open(self.train_data, "r") as f:
             self.num_of_training_docs = sum(1 for _ in f)
+        encoder, tokenizer = bert.load_bert(self)
```

Comment:
I would imagine having a light-weight config with only references to model names/paths instead of already loaded instances. Is there any specific reason to have it like that?

Reply:
Well, u r right. However, in order to create the training data, we need both an encoder and a tokenizer. 
I thought it is going to be a good idea to create a ModelBank abstraction that will have these submodels. These submodels are kinda also part of a config.
So, in my head `.toml` is a lightweight document with config data. Then `Config` dataclass is an abstaction that has configurable elements that will serve for `Coref` class. Im opened to change it but in the current set up I dont have much ideas how to split it. We can create and additional low prio task for that if thats an option

Reply:
I agree with the comment. If it works, it is okay for now, and we can tweak the thing in the future, as we already have a backlog item for that.